File: scripts/generate-images.py
#!/usr/bin/env python3
"""
DALL-E 3 Image Generator for Travel Blog
Generates unique, high-quality travel images for each article.
"""

import os
import json
import time
import urllib.request
import urllib.error
from pathlib import Path

API_KEY = os.environ.get("OPENAI_API_KEY", "")
if not API_KEY:
    # Try reading from .env file
    env_path = Path(__file__).parent.parent / ".env"
    if env_path.exists():
        for line in env_path.read_text().splitlines():
            if line.startswith("OPENAI_API_KEY="):
                API_KEY = line.split("=", 1)[1].strip()

OUTPUT_DIR = Path(__file__).parent.parent / "public" / "images" / "dubai"
OUTPUT_DIR.mkdir(parents=True, exist_ok=True)

# Each entry: (filename, DALL-E prompt, size)
IMAGE_SPECS = [
    # Site hero / general Dubai
    (
        "dubai-hero.webp",
        "A breathtaking wide panoramic photograph of the Dubai skyline at golden hour sunset, "
        "featuring the Burj Khalifa, modern skyscrapers, and the Dubai Marina reflecting on calm water. "
        "Warm orange and gold tones, professional travel photography, ultra high quality, cinematic lighting, "
        "no text, no watermarks, no logos",
        "1792x1024"
    ),
    # Complete travel guide
    (
        "dubai-travel-guide.webp",
        "An aerial photograph of Dubai showing the Burj Khalifa, Palm Jumeirah, and the coastline "
        "with turquoise Arabian Gulf water. Clear blue sky, vivid colors, professional drone photography "
        "style, travel magazine cover quality, no text, no watermarks",
        "1792x1024"
    ),
    # Budget vs luxury
    (
        "dubai-budget-vs-luxury.webp",
        "A split-composition photograph showing the contrast of Dubai: one side showing a vibrant "
        "traditional souk market with colorful spices and lanterns, the other side showing a luxurious "
        "gold-accented hotel lobby with chandeliers. Rich warm colors, professional photography, "
        "no text, no watermarks",
        "1792x1024"
    ),
    # Safety guide
    (
        "dubai-safety.webp",
        "A peaceful photograph of a well-lit Dubai promenade at twilight with families walking safely "
        "along the waterfront, modern city lights in the background, the Burj Al Arab visible in the "
        "distance, calm and secure atmosphere, professional travel photography, no text, no watermarks",
        "1792x1024"
    ),
    # Things to do
    (
        "dubai-things-to-do.webp",
        "A vibrant collage-style photograph showing exciting Dubai activities: a view from the top "
        "of Burj Khalifa observation deck looking down at the city, with a desert landscape visible "
        "in the distance, golden sunlight, adventurous and exciting mood, professional photography, "
        "no text, no watermarks",
        "1792x1024"
    ),
    # Desert safari
    (
        "dubai-desert-safari.webp",
        "A stunning photograph of a desert safari scene in the Dubai desert at sunset. Golden sand "
        "dunes stretching into the horizon, a luxury 4x4 vehicle driving up a steep dune, dramatic "
        "orange and purple sky, sand particles catching the light, professional adventure photography, "
        "no text, no watermarks",
        "1792x1024"
    ),
    # 3-day itinerary
    (
        "dubai-3-day-itinerary.webp",
        "A beautiful photograph of the Dubai Frame landmark with the old Dubai neighborhood on one "
        "side and the modern skyline on the other, taken during blue hour with city lights beginning "
        "to glow, perfect symmetry, professional architecture photography, no text, no watermarks",
        "1792x1024"
    ),
    # Food guide
    (
        "dubai-food.webp",
        "A top-down food photography shot of a beautifully arranged spread of Middle Eastern cuisine "
        "on an ornate brass tray: hummus, shawarma, kebabs, fresh flatbread, falafel, colorful mezze, "
        "baklava, Arabic coffee in traditional dallah. Warm ambient lighting, restaurant setting, "
        "professional food photography, no text, no watermarks",
        "1792x1024"
    ),
    # Shopping malls
    (
        "dubai-shopping.webp",
        "An impressive interior photograph of a grand Dubai shopping mall with multiple levels, "
        "a massive indoor waterfall feature, luxury brand storefronts with warm lighting, marble "
        "floors, and a dramatic glass ceiling letting in natural light, wide-angle perspective, "
        "professional interior photography, no text, no watermarks",
        "1792x1024"
    ),
    # Best hotels
    (
        "dubai-hotels.webp",
        "A luxurious photograph of a Dubai beachfront resort with an infinity pool overlooking "
        "the Arabian Gulf, palm trees, the Burj Al Arab in the background, plush sun loungers, "
        "crystal blue water, golden hour lighting, ultimate luxury atmosphere, professional hotel "
        "photography, no text, no watermarks",
        "1792x1024"
    ),
    # Currency guide
    (
        "dubai-currency.webp",
        "A close-up still life photograph of UAE Dirham banknotes and coins artistically arranged "
        "on a wooden surface with a traditional Arabic coffee cup, a small souk shopping bag, and "
        "a Dubai city map in the soft background, warm natural lighting, professional photography, "
        "no text overlays, no watermarks",
        "1792x1024"
    ),
    # Visa guide
    (
        "dubai-visa.webp",
        "A photograph of the modern Dubai International Airport arrivals hall with sleek architecture, "
        "glass facades, warm lighting, travelers walking through the impressive terminal, palm trees "
        "visible through floor-to-ceiling windows, welcoming atmosphere, professional travel photography, "
        "no text, no watermarks",
        "1792x1024"
    ),
    # Sidebar images (square format)
    (
        "dubai-sidebar-1.webp",
        "A vertical photograph of the Burj Khalifa tower at night with colorful LED light show "
        "reflecting in the Dubai Fountain water below, vibrant blues and purples, professional "
        "night photography, no text, no watermarks",
        "1024x1024"
    ),
    (
        "dubai-sidebar-2.webp",
        "A photograph of a traditional wooden abra boat on Dubai Creek at sunset with the gold "
        "souk market buildings glowing in warm light, reflections on calm water, cultural atmosphere, "
        "professional photography, no text, no watermarks",
        "1024x1024"
    ),
    (
        "dubai-sidebar-3.webp",
        "A photograph of the Palm Jumeirah from above showing the iconic palm tree shape island "
        "surrounded by turquoise water, luxury villas along the fronds, Atlantis hotel at the tip, "
        "clear sky, professional aerial photography, no text, no watermarks",
        "1024x1024"
    ),
]


def generate_image(prompt: str, size: str = "1792x1024") -> str | None:
    """Call DALL-E 3 API and return the image URL."""
    url = "https://api.openai.com/v1/images/generations"
    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {API_KEY}",
    }
    body = json.dumps({
        "model": "dall-e-3",
        "prompt": prompt,
        "n": 1,
        "size": size,
        "quality": "standard",
        "response_format": "url",
    }).encode("utf-8")

    req = urllib.request.Request(url, data=body, headers=headers, method="POST")

    try:
        with urllib.request.urlopen(req, timeout=120) as resp:
            data = json.loads(resp.read().decode("utf-8"))
            image_url = data["data"][0]["url"]
            revised_prompt = data["data"][0].get("revised_prompt", "")
            print(f"  Revised prompt: {revised_prompt[:100]}...")
            return image_url
    except urllib.error.HTTPError as e:
        error_body = e.read().decode("utf-8") if e.fp else "No details"
        print(f"  API Error {e.code}: {error_body[:200]}")
        return None
    except Exception as e:
        print(f"  Error: {e}")
        return None


def download_image(image_url: str, save_path: Path) -> bool:
    """Download image from URL and save to disk."""
    try:
        req = urllib.request.Request(image_url)
        with urllib.request.urlopen(req, timeout=60) as resp:
            save_path.write_bytes(resp.read())
        size_kb = save_path.stat().st_size / 1024
        print(f"  Saved: {save_path.name} ({size_kb:.0f} KB)")
        return True
    except Exception as e:
        print(f"  Download error: {e}")
        return False


def main():
    if not API_KEY:
        print("ERROR: No OPENAI_API_KEY found in environment or .env file")
        return

    print(f"Generating {len(IMAGE_SPECS)} images for Dubai travel blog...")
    print(f"Output directory: {OUTPUT_DIR}\n")

    results = {}
    for i, (filename, prompt, size) in enumerate(IMAGE_SPECS):
        save_path = OUTPUT_DIR / filename

        # Skip if already generated
        if save_path.exists() and save_path.stat().st_size > 10000:
            print(f"[{i+1}/{len(IMAGE_SPECS)}] SKIP (exists): {filename}")
            results[filename] = True
            continue

        print(f"[{i+1}/{len(IMAGE_SPECS)}] Generating: {filename}")
        print(f"  Size: {size}")

        image_url = generate_image(prompt, size)
        if image_url:
            success = download_image(image_url, save_path)
            results[filename] = success
        else:
            results[filename] = False
            print(f"  FAILED to generate {filename}")

        # Rate limiting: small pause between requests
        if i < len(IMAGE_SPECS) - 1:
            print("  Waiting 2s...")
            time.sleep(2)

    # Summary
    print("\n" + "=" * 50)
    print("GENERATION SUMMARY")
    print("=" * 50)
    success_count = sum(1 for v in results.values() if v)
    print(f"Success: {success_count}/{len(IMAGE_SPECS)}")
    for fname, ok in results.items():
        status = "OK" if ok else "FAILED"
        print(f"  [{status}] {fname}")


if __name__ == "__main__":
    main()

File: scripts/generate-images.ts
/**
 * DALL-E 3 Image Generator for Travel Blog
 * -----------------------------------------
 * Generates unique, high-quality AI images for each travel article.
 *
 * Usage:
 *   npm run generate:images                    # Generate all missing images
 *   npm run generate:images -- --city dubai     # Generate for specific city
 *   npm run generate:images -- --force          # Regenerate all (overwrite)
 *   npm run generate:images -- --dry-run        # Preview prompts without generating
 */

import 'dotenv/config';
import OpenAI from 'openai';
import fs from 'fs/promises';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const PROJECT_ROOT = path.resolve(__dirname, '..');
const PUBLIC_DIR = path.join(PROJECT_ROOT, 'public');
const CONTENT_DIR = path.join(PROJECT_ROOT, 'src', 'content', 'blog');

// ─── Configuration ───────────────────────────────────────────────────────────

const openai = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

const DALLE_CONFIG = {
  model: 'dall-e-3' as const,
  quality: 'standard' as const,
  style: 'natural' as const,     // "natural" for photos, "vivid" for artistic
  response_format: 'url' as const,
};

const IMAGE_SIZES = {
  hero: '1792x1024' as const,     // Landscape for hero banners
  article: '1792x1024' as const,  // Landscape for article headers
  sidebar: '1024x1024' as const,  // Square for sidebar galleries
  thumbnail: '1024x1024' as const,// Square for card thumbnails
};

// ─── Image Prompt Templates ─────────────────────────────────────────────────

interface ImageSpec {
  filename: string;
  prompt: string;
  size: keyof typeof IMAGE_SIZES;
  category: 'hero' | 'article' | 'sidebar';
}

/**
 * Generate image specs for a given city.
 * These prompts are crafted to produce stunning, text-free travel photography.
 */
function getCityImageSpecs(city: string, country: string): ImageSpec[] {
  const cityLower = city.toLowerCase().replace(/\s+/g, '-');

  // Base photography style suffix
  const photoStyle = 'Professional travel photography, ultra high quality, cinematic lighting, sharp focus, no text, no watermarks, no logos, no words, no letters, no numbers';

  return [
    // ─── Site Hero ───
    {
      filename: `${cityLower}-hero.webp`,
      prompt: `A breathtaking wide panoramic photograph of the ${city} skyline at golden hour sunset, featuring the most iconic landmarks and architecture, warm orange and gold tones reflecting on water or glass buildings. ${photoStyle}`,
      size: 'hero',
      category: 'hero',
    },

    // ─── Article Images (mapped to topic categories) ───
    {
      filename: `${cityLower}-travel-guide.webp`,
      prompt: `An aerial photograph of ${city}, ${country} showing the most famous landmarks from a drone perspective, with the coastline or landscape visible, vivid colors, clear sky, travel magazine cover quality. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },
    {
      filename: `${cityLower}-budget-vs-luxury.webp`,
      prompt: `A split-composition photograph showing the contrast of ${city}: one side showing a vibrant traditional local market with colorful goods and warm ambient lighting, the other side showing an ultra-luxurious hotel lobby or restaurant with gold accents and chandeliers, rich warm colors. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },
    {
      filename: `${cityLower}-safety.webp`,
      prompt: `A peaceful photograph of a well-lit ${city} promenade or boulevard at twilight with families and couples walking safely, modern city lights in the background, famous landmarks visible in the distance, calm and welcoming atmosphere. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },
    {
      filename: `${cityLower}-things-to-do.webp`,
      prompt: `A vibrant photograph from the top of the most famous observation point in ${city}, looking down at the sprawling city with parks, rivers or coastline visible, dramatic perspective showing the scale and beauty of the city, golden hour light. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },
    {
      filename: `${cityLower}-experiences.webp`,
      prompt: `A stunning action photograph of the most popular outdoor adventure or cultural experience in ${city}, ${country} — such as a desert safari, boat ride, cultural festival, or nature excursion — with dramatic lighting, vivid colors, and a sense of excitement. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },
    {
      filename: `${cityLower}-itinerary.webp`,
      prompt: `A beautiful photograph of the most iconic landmark in ${city} during blue hour with city lights glowing warmly, perfect symmetry, long exposure effect on any moving elements, professional architecture and cityscape photography. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },
    {
      filename: `${cityLower}-food.webp`,
      prompt: `A top-down food photography shot of a beautifully arranged spread of traditional ${city}, ${country} cuisine on an ornate table: local signature dishes, fresh bread, colorful sides, traditional beverages in authentic serving ware. Warm ambient restaurant lighting. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },
    {
      filename: `${cityLower}-shopping.webp`,
      prompt: `An impressive interior photograph of a famous shopping destination in ${city} — either a grand modern mall or a traditional market/souk — with beautiful architecture, warm lighting, and a sense of grandeur. Wide-angle perspective. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },
    {
      filename: `${cityLower}-hotels.webp`,
      prompt: `A luxurious photograph of a premium hotel in ${city} with a stunning view — infinity pool, rooftop terrace, or balcony overlooking the city skyline, plush furnishings, golden hour lighting, ultimate luxury travel atmosphere. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },
    {
      filename: `${cityLower}-currency.webp`,
      prompt: `A stylish still life photograph featuring local currency of ${country} alongside a cup of local traditional beverage, a small shopping bag, and a folded city map, on a textured wooden surface with soft natural side lighting, shallow depth of field. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },
    {
      filename: `${cityLower}-visa.webp`,
      prompt: `A modern photograph of the international airport or main transport hub in ${city} — sleek architecture, glass facades, warm lighting, travelers in a beautiful terminal building, welcoming atmosphere. ${photoStyle}`,
      size: 'article',
      category: 'article',
    },

    // ─── Sidebar Gallery Images ───
    {
      filename: `${cityLower}-sidebar-1.webp`,
      prompt: `A vertical photograph of the most iconic tower or tall building in ${city} at night with colorful lights and reflections in nearby water, vibrant blues and purples, professional night photography. ${photoStyle}`,
      size: 'sidebar',
      category: 'sidebar',
    },
    {
      filename: `${cityLower}-sidebar-2.webp`,
      prompt: `A photograph of a traditional cultural scene in ${city} — old quarter, traditional boats, historic market, or local craftsmen at work — at sunset with golden warm tones, cultural atmosphere. ${photoStyle}`,
      size: 'sidebar',
      category: 'sidebar',
    },
    {
      filename: `${cityLower}-sidebar-3.webp`,
      prompt: `A photograph of the most unique geographic or architectural feature near ${city} from an aerial perspective — an island, mountain, desert, river, or iconic bridge — surrounded by beautiful water or landscape, vivid colors, dramatic scale. ${photoStyle}`,
      size: 'sidebar',
      category: 'sidebar',
    },
  ];
}

// ─── Mapping: article filename → image filename ──────────────────────────────

/**
 * Maps markdown article filenames to their corresponding generated image.
 * This is used to update frontmatter heroImage fields.
 */
function getArticleImageMap(citySlug: string): Record<string, string> {
  return {
    [`complete-${citySlug}-travel-guide.md`]: `${citySlug}-travel-guide.webp`,
    [`${citySlug}-budget-vs-luxury-guide.md`]: `${citySlug}-budget-vs-luxury.webp`,
    [`${citySlug}-safety-guide.md`]: `${citySlug}-safety.webp`,
    [`top-things-to-do-${citySlug}.md`]: `${citySlug}-things-to-do.webp`,
    [`${citySlug}-desert-safari-guide.md`]: `${citySlug}-experiences.webp`,
    [`3-day-${citySlug}-itinerary.md`]: `${citySlug}-itinerary.webp`,
    [`what-to-eat-${citySlug}.md`]: `${citySlug}-food.webp`,
    [`best-shopping-malls-${citySlug}.md`]: `${citySlug}-shopping.webp`,
    [`best-hotels-${citySlug}.md`]: `${citySlug}-hotels.webp`,
    [`${citySlug}-currency-guide.md`]: `${citySlug}-currency.webp`,
    [`${citySlug}-visa-guide.md`]: `${citySlug}-visa.webp`,
  };
}

// ─── Image Generation ────────────────────────────────────────────────────────

async function generateImage(spec: ImageSpec): Promise<string | null> {
  const size = IMAGE_SIZES[spec.size];

  try {
    console.log(`  Calling DALL-E 3 (${size})...`);
    const response = await openai.images.generate({
      ...DALLE_CONFIG,
      prompt: spec.prompt,
      size: size,
      n: 1,
    });

    const imageUrl = response.data[0]?.url;
    const revisedPrompt = response.data[0]?.revised_prompt;

    if (revisedPrompt) {
      console.log(`  Revised: ${revisedPrompt.slice(0, 80)}...`);
    }

    return imageUrl || null;
  } catch (error: any) {
    console.error(`  API Error: ${error.message}`);
    if (error.code === 'content_policy_violation') {
      console.error('  Content policy violation — trying with simplified prompt...');
      // Retry with a simpler prompt
      try {
        const simplePrompt = `Beautiful travel photograph of a famous location, ${spec.prompt.split(',').slice(0, 2).join(',')}. Professional photography, no text.`;
        const retryResponse = await openai.images.generate({
          ...DALLE_CONFIG,
          prompt: simplePrompt,
          size: size,
          n: 1,
        });
        return retryResponse.data[0]?.url || null;
      } catch {
        return null;
      }
    }
    return null;
  }
}

async function downloadImage(url: string, savePath: string): Promise<boolean> {
  try {
    const response = await fetch(url);
    if (!response.ok) throw new Error(`HTTP ${response.status}`);
    const buffer = Buffer.from(await response.arrayBuffer());
    await fs.writeFile(savePath, buffer);
    const stats = await fs.stat(savePath);
    console.log(`  Saved: ${path.basename(savePath)} (${(stats.size / 1024).toFixed(0)} KB)`);
    return true;
  } catch (error: any) {
    console.error(`  Download error: ${error.message}`);
    return false;
  }
}

// ─── Markdown Updater ────────────────────────────────────────────────────────

async function updateMarkdownImages(city: string, citySlug: string): Promise<void> {
  const cityContentDir = path.join(CONTENT_DIR, citySlug);
  const imageMap = getArticleImageMap(citySlug);

  try {
    const files = await fs.readdir(cityContentDir);
    const mdFiles = files.filter(f => f.endsWith('.md'));

    for (const mdFile of mdFiles) {
      const imageName = imageMap[mdFile];
      if (!imageName) continue;

      const imagePath = `/images/${citySlug}/${imageName}`;
      const filePath = path.join(cityContentDir, mdFile);
      let content = await fs.readFile(filePath, 'utf-8');

      // Replace heroImage in frontmatter
      content = content.replace(
        /heroImage:\s*['"]?.*?['"]?\s*$/m,
        `heroImage: '${imagePath}'`
      );

      await fs.writeFile(filePath, content);
      console.log(`  Updated: ${mdFile} → ${imagePath}`);
    }
  } catch (error: any) {
    console.error(`  Error updating markdown: ${error.message}`);
  }
}

// ─── Main ────────────────────────────────────────────────────────────────────

async function main() {
  const args = process.argv.slice(2);
  const force = args.includes('--force');
  const dryRun = args.includes('--dry-run');
  const cityArg = args.find((_, i, a) => a[i - 1] === '--city') || 'dubai';

  // City configurations (expand this as you add more cities)
  const cities: Record<string, { name: string; country: string }> = {
    dubai: { name: 'Dubai', country: 'UAE' },
    paris: { name: 'Paris', country: 'France' },
    tokyo: { name: 'Tokyo', country: 'Japan' },
    london: { name: 'London', country: 'United Kingdom' },
    newyork: { name: 'New York', country: 'USA' },
    bali: { name: 'Bali', country: 'Indonesia' },
    rome: { name: 'Rome', country: 'Italy' },
    bangkok: { name: 'Bangkok', country: 'Thailand' },
    istanbul: { name: 'Istanbul', country: 'Turkey' },
    singapore: { name: 'Singapore', country: 'Singapore' },
  };

  const cityConfig = cities[cityArg.toLowerCase()];
  if (!cityConfig) {
    console.error(`Unknown city: ${cityArg}. Available: ${Object.keys(cities).join(', ')}`);
    process.exit(1);
  }

  const citySlug = cityArg.toLowerCase();
  const imageDir = path.join(PUBLIC_DIR, 'images', citySlug);
  await fs.mkdir(imageDir, { recursive: true });

  const specs = getCityImageSpecs(cityConfig.name, cityConfig.country);

  console.log(`\n🎨 DALL-E 3 Image Generator for ${cityConfig.name} Travel Blog`);
  console.log(`${'─'.repeat(55)}`);
  console.log(`City: ${cityConfig.name}, ${cityConfig.country}`);
  console.log(`Images to generate: ${specs.length}`);
  console.log(`Output: ${imageDir}`);
  console.log(`Mode: ${dryRun ? 'DRY RUN' : force ? 'FORCE (overwrite)' : 'Normal (skip existing)'}`);
  console.log(`${'─'.repeat(55)}\n`);

  if (dryRun) {
    specs.forEach((spec, i) => {
      console.log(`[${i + 1}/${specs.length}] ${spec.filename}`);
      console.log(`  Size: ${IMAGE_SIZES[spec.size]}`);
      console.log(`  Prompt: ${spec.prompt.slice(0, 120)}...`);
      console.log();
    });
    console.log('Dry run complete. No images generated.');
    return;
  }

  if (!process.env.OPENAI_API_KEY) {
    console.error('ERROR: OPENAI_API_KEY not found. Create a .env file with your key.');
    process.exit(1);
  }

  const results: { filename: string; success: boolean }[] = [];

  for (let i = 0; i < specs.length; i++) {
    const spec = specs[i];
    const savePath = path.join(imageDir, spec.filename);

    // Skip existing unless --force
    try {
      const stat = await fs.stat(savePath);
      if (!force && stat.size > 10000) {
        console.log(`[${i + 1}/${specs.length}] SKIP (exists): ${spec.filename}`);
        results.push({ filename: spec.filename, success: true });
        continue;
      }
    } catch {
      // File doesn't exist, generate it
    }

    console.log(`[${i + 1}/${specs.length}] Generating: ${spec.filename}`);
    const imageUrl = await generateImage(spec);

    if (imageUrl) {
      const success = await downloadImage(imageUrl, savePath);
      results.push({ filename: spec.filename, success });
    } else {
      results.push({ filename: spec.filename, success: false });
      console.log(`  FAILED`);
    }

    // Rate limiting: pause between API calls
    if (i < specs.length - 1) {
      console.log(`  Waiting 1.5s...\n`);
      await new Promise(r => setTimeout(r, 1500));
    }
  }

  // Update markdown files to point to generated images
  console.log(`\n📝 Updating markdown frontmatter...`);
  await updateMarkdownImages(cityConfig.name, citySlug);

  // Summary
  const successCount = results.filter(r => r.success).length;
  console.log(`\n${'═'.repeat(55)}`);
  console.log(`✅ GENERATION COMPLETE: ${successCount}/${results.length} images`);
  console.log(`${'═'.repeat(55)}`);
  results.forEach(r => {
    const icon = r.success ? '✅' : '❌';
    console.log(`  ${icon} ${r.filename}`);
  });

  if (successCount < results.length) {
    console.log(`\n⚠️  ${results.length - successCount} images failed. Run again to retry.`);
  }
}

main().catch(console.error);
